From 93fcc9eddab5c3e39385a526c924a7e7ed88d86f Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 4 Sep 2025 15:50:51 +0800
Subject: [PATCH 80/96] ia64: declare non-prototyped functions as static

Since kernel >= 6.7, non-static function declarations without prototypes
are no longer allowed.

Mark functions without prototypes as static to fix build except
uncached_alloc_page and uncached_free_page, since they are meant to be
exported globally.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Xinhui Yang <cyan@cyano.uk>
---
 arch/ia64/kernel/cyclone.c  | 2 +-
 arch/ia64/kernel/uncached.c | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/ia64/kernel/cyclone.c b/arch/ia64/kernel/cyclone.c
index 258d7b70c0f3..c81d614a1a24 100644
--- a/arch/ia64/kernel/cyclone.c
+++ b/arch/ia64/kernel/cyclone.c
@@ -35,7 +35,7 @@ static struct clocksource clocksource_cyclone = {
         .flags          = CLOCK_SOURCE_IS_CONTINUOUS,
 };
 
-int __init init_cyclone_clock(void)
+static int __init init_cyclone_clock(void)
 {
 	u64 __iomem *reg;
 	u64 base;	/* saved cyclone base address */
diff --git a/arch/ia64/kernel/uncached.c b/arch/ia64/kernel/uncached.c
index a0fec82c56b8..e8b65073e265 100644
--- a/arch/ia64/kernel/uncached.c
+++ b/arch/ia64/kernel/uncached.c
@@ -37,6 +37,8 @@ struct uncached_pool {
 
 struct uncached_pool uncached_pools[MAX_NUMNODES];
 
+unsigned long uncached_alloc_page(int starting_nid, int n_pages);
+void uncached_free_page(unsigned long uc_addr, int n_pages);
 
 static void uncached_ipi_visibility(void *data)
 {
-- 
2.25.1

